app/renderer/js/preload.ts

```diff
@@ -1,50 +1,61 @@
 'use strict';
 
-const { ipcRenderer, shell } = require('electron');
-const SetupSpellChecker = require('./spellchecker');
+import { ipcRenderer, shell } from 'electron';
+import SetupSpellChecker from './spellchecker';
 
-const LinkUtil = require(__dirname + '/utils/link-util.js');
-const params = require(__dirname + '/utils/params-util.js');
+import LinkUtil = require('./utils/link-util');
+import params = require('./utils/params-util');
+
+interface PatchedGlobal extends NodeJS.Global {
+	logout: () => void;
+	shortcut: () => void;
+}
+
+const globalPatched = global as PatchedGlobal;
 
 // eslint-disable-next-line import/no-unassigned-import
 require('./notification');
 
 // Prevent drag and drop event in main process which prevents remote code executaion
 require(__dirname + '/shared/preventdrag.js');
 
-// eslint-disable-next-line camelcase
+declare let window: ZulipWebWindow;
```

Comment:
What's the use case of this?

Reply:
We need to declare the window variable before using it. This type is declared in typings.d.ts file.

Reply:
Got it.